nx CLI changes forward slashes in package name arguments to backslashes on Windows

## Current Behavior
The nx CLI changes forward slashes in package name arguments to backslashes on Windows.

Accordingly, it is not possible to run commands such as:
- `nx migrate @nrwl/workspace`
- `nx update @angular/cli`

as the package name gets converted to e.g. '@nrwl\workspace' or '@angular\cli' in the underlying command - see 'Failure Logs' below for more details.

## Expected Behavior
nx should not rewrite forward slashes in package name arguments as backslashes on Windows.

## Steps to Reproduce

1. Install NodeJS LTS on Windows 10 (currently 14.15.0)
2. Open a PowerShell window in order to enter the subsequent commands
3. Install nx globally via `npm install -g nx`
4. Create a new nx workspace using `npx create-nx-workspace nxtest`
5. Run `nx migrate @nrwl/workspace` from within the new nx workspace directory

### Failure Logs

Note the references to `@nrwl\workspace` in the logs below (rather than `@nrwl/workspace` as was typed in the original command):
```
PS C:\Users\<username>\source\repos\<projectname>> nx migrate @nrwl/workspace
Fetching meta data about packages.
It may take a few minutes.
Fetching @nrwl\workspace@latest

 ERROR  The migrate command failed. Try the following to migrate your workspace:

> npm install --save-dev @nrwl/workspace@latest
> nx migrate @nrwl\workspace@latest --from="@nrwl/workspace@10.3.2"
This will use the newest version of the migrate functionality, which might have your issue resolved.
----------------------------------------------------------------------------------------------------
Command failed: npm install @nrwl\workspace@latest --prefix=C:\Users\<USERNAME>~1\AppData\Local\Temp\tmp-27228eZpIVvNHv50i
npm ERR! code EINVALIDPACKAGENAME
npm ERR! Invalid package name "@nrwl\workspace": name can only contain URL-friendly characters

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\<username>\AppData\Local\npm-cache\_logs\2020-11-06T11_30_19_852Z-debug.log

npm ERR! code 1
npm ERR! path C:\Users\<username>\source\repos\illumin8-frontend
npm ERR! command failed
npm ERR! command C:\WINDOWS\system32\cmd.exe /d /s /c "tao "migrate" "@nrwl\workspace""

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\<username>\AppData\Local\npm-cache\_logs\2020-11-06T11_30_19_902Z-debug.log

>  NX   ERROR  Could not find @nrwl/cli module in this workspace.

  Have you run npm/yarn install?
```

### Environment
Windows 10 Pro
Version 10.0.19042 Build 19042
PowerShell version: 5.1.19041.610
NodeJS version: 14.15.0

nx report output:
```
  nx : Not Found
  @nrwl/angular : Not Found
  @nrwl/cli : 10.3.2
  @nrwl/cypress : Not Found
  @nrwl/eslint-plugin-nx : Not Found
  @nrwl/express : Not Found
  @nrwl/jest : Not Found
  @nrwl/linter : Not Found
  @nrwl/nest : Not Found
  @nrwl/next : Not Found
  @nrwl/node : Not Found
  @nrwl/react : Not Found
  @nrwl/schematics : Not Found
  @nrwl/tao : 10.3.2
  @nrwl/web : Not Found
  @nrwl/workspace : 10.3.2
  typescript : 4.0.5
```

--- a/packages/tao/src/commands/migrate.spec.ts
+++ b/packages/tao/src/commands/migrate.spec.ts
@@ -542,5 +542,27 @@ describe('Migration', () => {
         parseMigrationsOptions(['8.12.0', '--to', 'myscope'])
       ).toThrowError(`Incorrect 'to' section. Use --to="package@version"`);
     });
+
+    it('should handle backslashes in package names', () => {
+      const r = parseMigrationsOptions([
+        '@nrwl\\workspace@8.12.0',
+        '--from',
+        '@myscope\\a@12.3,@myscope\\b@1.1.1',
+        '--to',
+        '@myscope\\c@12.3.1',
+      ]);
+      expect(r).toEqual({
+        type: 'generateMigrations',
+        targetPackage: '@nrwl/workspace',
+        targetVersion: '8.12.0',
+        from: {
+          '@myscope/a': '12.3.0',
+          '@myscope/b': '1.1.1',
+        },
+        to: {
+          '@myscope/c': '12.3.1',
+        },
+      });
+    });
   });
 });

--- a/packages/tao/src/commands/migrate.ts
+++ b/packages/tao/src/commands/migrate.ts
@@ -69,6 +69,10 @@ export function normalizeVersion(version: string) {
   }
 }
 
+function slash(packageName) {
+  return packageName.replace(/\\/g, '/');
+}
+
 export class Migrator {
   private readonly versions: (p: string) => string;
   private readonly fetch: (p: string, v: string) => Promise<MigrationsJson>;
@@ -304,7 +308,7 @@ function versionOverrides(overrides: string, param: string) {
         `Incorrect '${param}' section. Use --${param}="package@version"`
       );
     }
-    res[selectedPackage] = normalizeVersionWithTagCheck(selectedVersion);
+    res[slash(selectedPackage)] = normalizeVersionWithTagCheck(selectedVersion);
   });
   return res;
 }
@@ -378,7 +382,7 @@ export function parseMigrationsOptions(
     );
     return {
       type: 'generateMigrations',
-      targetPackage,
+      targetPackage: slash(targetPackage),
       targetVersion,
       from,
       to,

